ivy_tests/test_ivy/test_frontends/test_mindspore/test_ops/test_function/test_mindspore_nn_func.py

```diff
@@ -77,3 +77,42 @@
 #         on_device=on_device,
 #         x=x[0],
 #     )
+
+# adaptive_maxpool2d
+# @handle_frontend_test(
+#     fn_tree="mindspore.ops.function.nn_func.adaptive_maxpool2d",
+#     d_type_and_x=helpers.dtype_and_values(
+#         available_dtypes=helpers.get_dtypes("valid"),
+#         num_arrays=1,
+#         shared_dtype=True,
+#         min_value=2,
+#         max_value=5,
+#         min_dim_size=4,
+#         shape=(
+#             st.integers(min_value=2, max_value=10),
+#             4,
+#             st.integers(min_value=12, max_value=64),
+#             st.integers(min_value=12, max_value=64),
+#         ),
+#     ),
+#     output_size=st.tuples(st.integers(min_value=1, max_value=10), st.integers(min_value=1, max_value=10)),
+# )
+# def test_mindspore_adaptive_maxpool2d(
+#     *,
+#     d_type_and_x,
+#     output_size,
+#     on_device,
+#     fn_tree,
+#     frontend,
+#     test_flags,
+# ):
+#     dtype, x = d_type_and_x
+#     helpers.test_frontend_function(
+#         input_dtypes=dtype,
+#         frontend=frontend,
+#         test_flags=test_flags,
+#         fn_tree=fn_tree,
+#         on_device=on_device,
+#         input=x[0],
+#         output_size=output_size,
+#     )
```

Comment:
In order for me to merge the PR, I'll need you to generate a set of inputs using the strategies you have defined in your test implementation. Run your function using these inputs and also pass these inputs to the native mindspore function itself and then compare the outputs for the two. If the outputs are identical for both the functions for a set of inputs then it would be safe to assume that your implementation is correct. If not, then you would need to modify your implementation such that it returns the correct output. Also generate booleans for `return_indices` parameter. Kindly attach screenshots of this in the conversation tab once done so I can go through them and merge the PR afterwards. Thanks!